### ISSUE TITLE
loss does not decrease in Google Colab and local computer. (TF.Keras 1.13 vs from TF.Keras 1.14 to TF.Keras 2.6)

### ISSUE THREAD

**Issue by HeewonChung92** [keras-team/tf-keras#579]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Performance

### Source

source

### Tensorflow Version

1.13 vs 2.6

### Custom Code

Yes

### OS Platform and Distribution

Windows 10

### Python version

3.7.4

### CUDA/cuDNN version

11.2.0 / 8.1.0

### Current Behaviour?
I asked this issue to tensorflow, but tensorflow told me to ask keras repo.
https://github.com/tensorflow/tensorflow/issues/56080

This is my full source code!
https://colab.research.google.com/drive/1tSE7t3GPa8r4x7F4sPgj8CqiYqtOrej5?usp=sharing
https://drive.google.com/drive/folders/17Nnl5FcrUWnRgzz4DM_fNAI3KhwqcK3r?usp=sharing

```shell
I studied with my data and tf-gpu 1.6 and keras in 2019. 
At this time, training was done normally.

Although it is the same code as in the past, 
training loss and validation loss are decreased 
when training with TF 1.13  (working TF 1.6 ~ TF 1.13).

However, from TF 1.14 to TF 2.6, 
both train loss and validation loss have increased.

When the model weight made with TF 1.6 was called from TF 2.6 (tried to TF 1.14, TF 1.15, TF 2.0 and TF 2.6), 
it was called without any problems and the results were the same as those from the past.

So I can't understand why that happen.
I wonder what parts of TF 1.13 and TF 1.14 and above are different and affect training!!!!!!

Also, I tried updating the source code through tf_upgrade_v2, 
but it was not solved at all.
https://www.tensorflow.org/guide/upgrade?hl=en
tf_upgrade_v2 --infile my_code.py --outfile my_code_upgraded.py
```

### Relevant log output

**decreased the loss when using TF 1.13**
![image](https://user-images.githubusercontent.com/39148033/168065098-12939b28-dbe7-4cef-9cb3-ecb0ea18843c.png)

**increased the loss when using TF 1.14 and TF 2.6**
![image](https://user-images.githubusercontent.com/39148033/168066373-cb5bb8b4-1f18-41ba-a412-446c5d0dd195.png)

_No response_</details></details>
